### PR TITLE
Search User implementation

### LINKED ISSUE #281: [Frontend][Backend]Search User Feature
The Sidebar in Admin has the _Search User_ Button, but now way to actually search a user.

Screenshot 2022-06-28 at 1 59 59 PM

It still lacks the ability to find a user among the list of users there.

### PR DESCRIPTION
**Pull request by PrerakMathur20**
# Description

Added Search User functionality to the **Search User** page, with multiple new functionalities, such as -
* Download `.csv` list of all users
* Print `.pdf` of all users
* **Show** or **Hide** Columns
* Conditional Column Filters

Fixes #281 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All fields were checked and searched for as per the search query.

Also, include screenshots for the verification and reviewing purpose.
Screenshot 2022-07-25 at 7 31 53 AM

### Demo
https://user-images.githubusercontent.com/76054330/180678913-087d41d1-2d92-4203-b568-65ee5fefcf7b.mov

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works **(N/A)**
- [ ] New and existing unit tests pass locally with my changes **(N/A)**
- [ ] Any dependent changes have been merged and published in downstream modules **(N/A)**

### CODE REVIEW

File: React-frontend/src/components/Admin/SearchUserModel.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+             }
+    
+             // Redireect to member deails
+             history.push("/list-members/member/" + gridCellParam.id);
+         }
+    
+         // call handleCellClick for each row on double click
+         columns.forEach((column) => {
+             column.options.customBodyRender = (data, type, row) => {
+                 console.log("data, type, row", data, type, row);
```
Comment (swapnalshahil) [inline React-frontend/src/components/Admin/SearchUserModel.js]: Comment this line (console.log)!

File: React-frontend/src/components/Admin/SearchUserModel.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                     role: member.role,
+                 }))
+             );
+    
+         // Collection of all rows
+         const newRows = allRows.map((row) => {
+             return Object.values(row);
+         });
+    
+         console.log(newRows);
```
Comment (swapnalshahil) [inline React-frontend/src/components/Admin/SearchUserModel.js]: This one also!

**Review by swapnalshahil — CHANGES_REQUESTED** [review]

**Comment by PrerakMathur20** [comment]
Since those lines were unnecessary and useless as comments, so I removed them.

**Review by swapnalshahil — APPROVED** [review]
LGTM!

**Pull request merged**